src/native/sync/omni.rs

```diff
@@ -191,24 +192,24 @@ impl OmniBucket {
         local_deletion: bool,
         mut remote_deletion: bool,
     ) -> Result<String, NativeError> {
-        if let Ok(local) = self.get_local()
-            && local_deletion
-        {
-            local.remove_data()?;
-            // Find index of bucket
-            let index = global
-                .buckets
-                .iter()
-                .position(|b| b == &local)
-                .ok_or(NativeError::missing_local_drive())?;
-            // Remove bucket config from global config
-            global.buckets.remove(index);
+        if local_deletion {
+            if let Ok(local) = self.get_local() {
+                local.remove_data()?;
+                // Find index of bucket
+                let index = global
+                    .buckets
+                    .iter()
+                    .position(|b| b == &local)
+                    .ok_or(NativeError::missing_local_drive())?;
+                // Remove bucket config from global config
+                global.buckets.remove(index);
+            }
         }
 
-        if let Ok(remote) = self.get_remote()
-            && remote_deletion
-        {
-            remote_deletion = RemoteBucket::delete_by_id(client, remote.id).await.is_ok();
+        if remote_deletion {
+            if let Ok(remote) = self.get_remote() {
+                remote_deletion = RemoteBucket::delete_by_id(client, remote.id).await.is_ok();
+            }
```

Comment:
This looks like another silent failure case. The remote not being present should probably be an error here. If the user of the library is expecting to remotely make a change, then they expect it to exist. Assumptions made by user and developers consuming a library assumptions should be explicitly managed through communication.